examples/no_std_macros.rs

```diff
@@ -0,0 +1,45 @@
+#![no_std]
+
+use hecs::{Bundle, Query};
+
+#[derive(Bundle)]
+pub struct Foo {
+    pub foo: i32,
+}
+
+#[derive(Bundle)]
+pub struct Bar {
+    pub foo: i32,
+}
+
+#[derive(Bundle)]
+pub struct Baz {
+    pub foo: i32,
+    pub baz: &'static str,
+}
+
+#[derive(Query)]
+pub struct Quux<'a> {
+    pub foo: &'a i32,
+    pub bar: &'a mut bool,
+}
+
+#[panic_handler]
+#[cfg(target_os = "none")]
+fn panic(_: &core::panic::PanicInfo) -> ! {
+    loop {}
+}
+
+struct NoAlloc;
```

Comment:
Could we have `extern crate alloc;` rather than this stubbed machinery? hecs won't work without dynamic allocation, so this might cause some confusion.

Reply:
Removed.

This is necessary for a `staticlib` build. `extern crate alloc;` does not solve the problem. But looks like a `rlib` build is already enough. Any use of the `::std` in the macro expansion will fail the no_std rlib build.